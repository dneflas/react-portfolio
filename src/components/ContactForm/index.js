import React, { useState } from "react";
import { validateEmail } from "../../utils/helpers";

function ContactForm() {
  const [errorMessage, setErrorMessage] = useState("");

  const [formState, setFormState] = useState({
    name: "",
    email: "",
    message: "",
  });

  const [contactLinks] = useState([
    {
      name: "Email",
      link: "mailto:debbieneflas@gmail.com",
      icon: "fa-solid fa-envelope",
    },
    {
      name: "Phone",
      link: "tel:+16263771363",
      icon: "fa-solid fa-phone",
    },
  ]);

  const { name, email, message } = formState;

  function handleChange(event) {
    if (event.target.name === "Email") {
      const isValid = validateEmail(event.target.value);
      if (!isValid) {
        setErrorMessage("Your email is invalid");
      } else {
        setErrorMessage("");
      }
    } else {
      if (!event.target.value.length) {
        setErrorMessage(`${event.target.name} is required.`);
      } else {
        setErrorMessage("");
      }
    }
    if (!errorMessage) {
      setFormState({ ...formState, [event.target.name]: event.target.value });
    }
  }

  function handleSubmit(event) {
    event.preventDefault();
    console.log(formState);

    setFormState({
      name: "",
      email: "",
      message: "",
    });
  }
  return (
    <section className="my-5">
      <h2>Contact Me</h2>
      <div className="flex-row my-3">
        <form className="col-12 col-md-8" onSubmit={handleSubmit}>
          <div className="my-1">
            <label className="form-label" htmlFor="Name">
              Name:{" "}
            </label>
            <input
              className="form-input"
              type="text"
              name="name"
              defaultValue={name}
              onBlur={handleChange}
            />
          </div>
          <div className="my-1">
            <label className="form-label" htmlFor="Email">
              Email:{" "}
            </label>
            <input
              className="form-input"
              type="text"
              name="email"
              defaultValue={email}
              onBlur={handleChange}
            />
          </div>
          <div className="my-1">
            <label className="form-label" htmlFor="Message">
              Message:{" "}
            </label>
            <textarea
              className="form-textarea"
              name="message"
              rows="8"
              defaultValue={message}
              onBlur={handleChange}
            />
          </div>
          {errorMessage && (
            <div>
              <p>{errorMessage}</p>
            </div>
          )}
          <button className="btn my-1" type="submit">
            Submit
          </button>
        </form>
        <div className="col-12 col-md-4 align-center my-auto">
          {contactLinks.map((link) => (
            <div className="col-2 text-center my-4" key={link.name}>
              <a
                href={link.link}
                className="ml-2 my-1 px-2 py-1 text-dark icons"
              >
                <i className={link.icon}></i>
                <span> : {link.name}</span>
              </a>
            </div>
          ))}
        </div>
      </div>
    </section>
  );
}

export default ContactForm;
