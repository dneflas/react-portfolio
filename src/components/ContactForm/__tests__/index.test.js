import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import ContactForm from "..";

afterEach(cleanup);

describe("Footer components render", () => {
  it("renders", () => {
    render(<ContactForm />);
  });
  it("matches snapshot", () => {
    const { asFragment } = render(<ContactForm />);
    expect(asFragment()).toMatchSnapshot();
  });
});
