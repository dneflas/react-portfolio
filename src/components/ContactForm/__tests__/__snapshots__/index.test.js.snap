// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer components render matches snapshot 1`] = `
<DocumentFragment>
  <section>
    <h2>
      Contact Me
    </h2>
    <div
      class="flex-row my-3"
    >
      <form
        class="col-12 col-md-8"
      >
        <div
          class="my-1"
        >
          <label
            class="form-label"
            for="Name"
          >
            Name: 
          </label>
          <input
            class="form-input"
            name="Name"
            type="text"
            value=""
          />
        </div>
        <div
          class="my-1"
        >
          <label
            class="form-label"
            for="Email"
          >
            Email: 
          </label>
          <input
            class="form-input"
            name="Email"
            type="text"
            value=""
          />
        </div>
        <div
          class="my-1"
        >
          <label
            class="form-label"
            for="Message"
          >
            Message: 
          </label>
          <textarea
            class="form-textarea"
            name="Message"
            rows="8"
          />
        </div>
        <button
          class="btn my-1"
          type="submit"
        >
          Submit
        </button>
      </form>
      <div
        class="col-12 col-md-4 align-center my-auto"
      >
        <div
          class="col-2 text-center my-5"
        >
          <a
            class="ml-2 my-1 px-2 py-1 text-dark icons"
            href="mailto:debbieneflas@gmail.com"
          >
            <i
              class="fa-solid fa-envelope"
            />
            <span>
               : Email
            </span>
          </a>
        </div>
        <div
          class="col-2 text-center my-5"
        >
          <a
            class="ml-2 my-1 px-2 py-1 text-dark icons"
            href="tel:+16263771363"
          >
            <i
              class="fa-solid fa-phone"
            />
            <span>
               : Phone
            </span>
          </a>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
